src/test/resources/regressions/features/hyper_properties/declassify-simple2.gobra

```diff
@@ -0,0 +1,295 @@
+// Declassify can only be called on a MAC tag of the most recently received message
+// generated with the private key.
+ghost
+decreases
+pure func (MWE1) Guard(s state, a Action) bool {
+	return match a {
+		// NOTE: This makes our IOD spec well-formed, as the previous I/O action
+		// determinizes what is allowed to be declassified.
+		case DeclassifyIO{?t}: t == MAC(s.key, s.lastMsg)
+		case _: true
+	}
+}
+
+ghost
+decreases
+pure func (MWE1) Update(s state, a Action) state {
+	return match a {
+		case RecvIO{?m}: state { key: s.key, lastMsg: m }
+		case _: s
+	}
+}
+
+////////////////////////////// Trusted library I/O //////////////////////////////
+
+// Receive message `m` over network.
+decreases
+requires acc(History())
+ensures acc(History()) && *History() == Snoc{old(*History()), RecvIO{m}}
+func Recv() (m int)
+
+// Send tag `t` over network.
+decreases
+requires acc(History())
+ensures acc(History()) && *History() == Snoc{old(*History()), SendIO{t}}
+func Send(t int)
+
+// Declassify tag `t`.
+ghost
+decreases
+requires acc(History())
+ensures acc(History()) && *History() == Snoc{old(*History()), DeclassifyIO{t}}
+func Declassify(t int)
+
+
+////////////////////////////// Program //////////////////////////////
+
+// Abstract function representing the computation of a MAC.
+// key x message -> MAC tag
+decreases
+pure func MAC(int, int) int
+
+// Receives a message, authenticates it using a MAC, and sends the resulting tag.
+// The state `s` contains the private key of this router, and the most recently received message.
+requires acc(History()) && acc(s) && low(Pre(DefaultClassification{}, *History())) && Reaches(MWE1{}, *History(), *s)
+func authenticate(s *state) {
+	invariant acc(History()) && acc(s) && low(Pre(DefaultClassification{}, *History())) && Reaches(MWE1{}, *History(), *s)
+	for {
+		m := Recv()
+		LowPost(DefaultClassification{})
+		ghost s.lastMsg = m
+		t := MAC(s.key, m)
+		ghost Declassify(t)
+		ghost LowPost(DefaultClassification{})
+		Send(t)
+	}
+}
+
+requires acc(History()) && acc(s) && low(Pre(DefaultClassification{}, *History())) && Reaches(MWE1{}, *History(), *s)
+func declMissingTest(s *state) {
+	invariant acc(History()) && acc(s) && low(Pre(DefaultClassification{}, *History())) && Reaches(MWE1{}, *History(), *s)
+	for {
+		m := Recv()
+		LowPost(DefaultClassification{})
+		ghost s.lastMsg = m
+		t := MAC(s.key, m)
+		// message not declassified
+		Send(t)
+		//:: ExpectedOutput(precondition_error)
+		ghost LowPost(DefaultClassification{})
+	}
+}
+
+requires acc(History()) && acc(s) && low(Pre(DefaultClassification{}, *History())) && Reaches(MWE1{}, *History(), *s)
+func declMissingLowPost(s *state) {
+	invariant acc(History()) && acc(s) && low(Pre(DefaultClassification{}, *History())) && Reaches(MWE1{}, *History(), *s)
+	for {
+		m := Recv()
+		LowPost(DefaultClassification{})
+		ghost s.lastMsg = m
+		t := MAC(s.key, m)
+		ghost Declassify(t)
+		// call to lowpost missing here
```

Comment:
Isn't this comment meant to point out the contrast to the other func above? Also, what does lowpost excatly do?